sources/api/controllers/GoogleCalendarController.ts

```diff
@@ -0,0 +1,43 @@
+import { Request, Response } from "express";
+import { Connection } from "typeorm";
+import { Credentials } from "google-auth-library";
+
+import { GoogleCalendarUtils } from "../../utils/googleCalendarUtils";
+import { getRequestingUser } from "../middleware/loggedOnly";
+import { DbConnection } from "../../utils/DbConnection";
+import { GoogleCalendarCredentials as GoogleCalendarCredentials } from "../../model/entity/GoogleCalendarCredentials";
+import { Calendar } from "../../model/entity/Calendar";
+
+
+export async function authorizeGoogleCalendar(req: Request, res: Response) {
+    const url: string = GoogleCalendarUtils.getConsentPageUrl();
+    res.redirect(url);    
+}
+
+export async function importGoogleCalendar(req: Request, res: Response) {
+    const user = getRequestingUser(req);
+
+    const code: string = req.query.code;
+    const token = await GoogleCalendarUtils.getTokenFromCode(code);
+    const credentials : Credentials = token.res.data;
+
+    const conn: Connection = await DbConnection.getConnection();
+   
+    const googleCalendarCredentials : GoogleCalendarCredentials = new GoogleCalendarCredentials(
+        credentials.access_token, credentials.refresh_token, 
+        "", credentials.token_type, new Date(credentials.expiry_date),
+        user
+    );
+
+    await conn.manager.save(googleCalendarCredentials);
+
+    const calendars: Calendar[] = await GoogleCalendarUtils.getCalendarAndEventsFromCredentials(googleCalendarCredentials);
+    
+    calendars.forEach(async (cal)=> {
+        await Calendar.createCalendar(conn, cal, [user]);
+        cal.events.forEach((e) => { conn.manager.save(e); })
+    });
```

Comment:
Not a good idea to give async function to `forEach`

Prefer doing something more like
```ts
const calendarInsertions = calendars.map(c => return Calendar.create(c));
await Promise.all(calendarInsertions);
```